File: 卡码/代码随想录/1/09-动态规划/Go/02-背包问题/02-完全背包/Kama_0052.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, v int
	fmt.Scan(&n, &v)
	weight, value := make([]int, n), make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&weight[i])
		fmt.Scan(&value[i])
	}
	dp := make([]int, v+1)
	for i := 0; i < v+1; i++ {
		dp[i] = 0
	}
	for i := 0; i < n; i++ {
		for j := weight[i]; j < v+1; j++ {
			dp[j] = int(math.Max(float64(dp[j]), float64(dp[j-weight[i]]+value[i])))
		}
	}
	fmt.Println(dp[v])
}

File: 卡码/代码随想录/1/09-动态规划/Js/02-背包问题/02-完全背包/Kama_0052.js
//52. 携带研究材料
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

(async function(){
	const [n,v]=(await readline()).split(" ").map(Number);
	const weight=new Array(n);
	const value=new Array(n);
	for(let i=0;i<n;i++){
		[weight[i],value[i]]=(await readline()).split(" ").map(Number);
	}
	const dp=new Array(v+1).fill(0);
	for(let i=0;i<n;i++){
		for(let j=weight[i];j<v+1;j++){
			dp[j]=Math.max(dp[j],dp[j-weight[i]]+value[i]);
		}
	}
	console.log(dp[v]);
})();